assets/src/components/propertiesPanel/minischedule.tsx

```diff
@@ -61,14 +74,43 @@ export const MinischeduleBlock = ({ activeTripId }: Props): ReactElement => {
     return (
       <>
         <Header label="Block" value={block.id} />
-        {block.pieces.map((piece) => (
-          <Piece piece={piece} view="block" key={piece.start.time} />
-        ))}
+        <ActivityWrapper
+          activities={block.pieces}
+          vehicleOrGhost={vehicleOrGhost}
+          view="block"
+        />
       </>
     )
   }
 }
 
+const ActivityWrapper = ({
```

Comment:
I see how this deduplicates the `.map()` and `<Piece/>` calls, but is the extra layer of indirection worth it?

Reply:
This component was much more important in a previous iteration of the code, I think it can probably come out now that you mention it.